adapters/adhese/adhese.go

```diff
@@ -18,267 +14,197 @@ import (
-	} else {
-		bidResponse = convertAdheseOpenRtbBid(adheseBid)
-	}
+	return []*adapters.RequestData{requestData}, nil
+}
 
-	price, err := strconv.ParseFloat(adheseBid.Extension.Prebid.Cpm.Amount, 64)
-	if err != nil {
-		return nil, []error{err, WrapServerError(fmt.Sprintf("Could not parse Price %v as float ", string(adheseBid.Extension.Prebid.Cpm.Amount)))}
+func inferBidTypeFromImp(i openrtb2.Imp) (openrtb_ext.BidType, []error) {
+	if i.Banner != nil {
+		return openrtb_ext.BidTypeBanner, nil
 	}
-	width, err := strconv.ParseInt(adheseBid.Width, 10, 64)
-	if err != nil {
-		return nil, []error{err, WrapServerError(fmt.Sprintf("Could not parse Width %v as int ", string(adheseBid.Width)))}
+	if i.Video != nil {
+		return openrtb_ext.BidTypeVideo, nil
 	}
-	height, err := strconv.ParseInt(adheseBid.Height, 10, 64)
-	if err != nil {
-		return nil, []error{err, WrapServerError(fmt.Sprintf("Could not parse Height %v as int ", string(adheseBid.Height)))}
+	if i.Native != nil {
+		return openrtb_ext.BidTypeNative, nil
 	}
-	bidResponse.Cur = adheseBid.Extension.Prebid.Cpm.Currency
-	if len(bidResponse.SeatBid) > 0 && len(bidResponse.SeatBid[0].Bid) > 0 {
-		bidResponse.SeatBid[0].Bid[0].Price = price
-		bidResponse.SeatBid[0].Bid[0].W = width
-		bidResponse.SeatBid[0].Bid[0].H = height
+	if i.Audio != nil {
+		return openrtb_ext.BidTypeAudio, nil
 	}
 
-	bidderResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+	return "", []error{&errortypes.BadServerResponse{Message: "Could not infer bid type from imp"}}
+}
 
-	if len(bidResponse.SeatBid) == 0 {
-		return nil, []error{WrapServerError("Response resulted in an empty seatBid array.")}
-	}
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
 
-	var errs []error
-	for _, sb := range bidResponse.SeatBid {
-		for i := 0; i < len(sb.Bid); i++ {
-			bid := sb.Bid[i]
-			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
-				Bid:     &bid,
-				BidType: getBidType(bid.AdM),
-			})
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
 
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
 		}
+		return nil, []error{err}
 	}
-	return bidderResponse, errs
-}
 
-func convertAdheseBid(adheseBid AdheseBid, adheseExt AdheseExt, adheseOriginData AdheseOriginData) openrtb2.BidResponse {
-	adheseExtJson, err := json.Marshal(adheseOriginData)
-	if err != nil {
-		adheseExtJson = make([]byte, 0)
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
 	}
```

Comment:
could make use of response util here

https://github.com/prebid/prebid-server/blob/ec729e6e5d50a1306d76332fda52108b9d313c1d/adapters/response.go#L1-L28

Reply:
done!